scylla-cql/src/serialize/value.rs

```diff
@@ -471,13 +471,37 @@ impl<T: SerializeValue> SerializeValue for Vec<T> {
         typ: &ColumnType,
         writer: CellWriter<'b>,
     ) -> Result<WrittenCellProof<'b>, SerializationError> {
-        serialize_sequence(
-            std::any::type_name::<Self>(),
-            self.len(),
-            self.iter(),
-            typ,
-            writer,
-        )
+        match typ {
+            ColumnType::Collection {
+                typ: CollectionType::List(_) | CollectionType::Set(_),
+                ..
+            } => serialize_sequence(
+                std::any::type_name::<Self>(),
+                self.len(),
+                self.iter(),
+                typ,
+                writer,
+            ),
+
+            ColumnType::Vector {
+                typ: element_type,
+                dimensions,
+            } => serialize_vector(
+                std::any::type_name::<Self>(),
+                self.len(),
+                self.iter(),
+                element_type,
+                dimensions,
+                typ,
+                writer,
+            ),
+
+            _ => Err(mk_typck_err_named(
+                std::any::type_name::<Self>(),
+                typ,
+                SetOrListTypeCheckErrorKind::NotSetOrList,
+            )),
```

Comment:
:thinking: The name is no longer correct. It should be `NotSetOrListOrVector`, but we cannot change the name - this would break the API. Perhaps we should introduce this new variant, leaving the old one unused? cc @Lorak-mmk

Reply:
I don't think the name is important enough to justify such change

Reply:
OK, then let's update the docstring of that variant by mentioning Vector there.